src/Model/Behavior/VersionBehavior.php

```diff
@@ -161,6 +162,13 @@ public function beforeSave(Event $event, Entity $entity, ArrayObject $options)
                 'content' => $content,
                 'created' => $created,
             ];
+
+            $event = new Event('Model.Version.beforeSave', $this, $options);
+            $user_data = EventManager::instance()->dispatch($event);
+            if (isset($user_data->result) && is_array($user_data->result)) {
+                $data = array_merge($data, $user_data->result);
```

Comment:
We should note that the custom data will be merged with the versioned data, and as such people should be careful about what they merge in.